examples/esp-chat-react/js/model/eventProcessors/ChatAppEventProcessor.js

```diff
@@ -0,0 +1,60 @@
+"use strict";
+
+var esp = require('esp-js');
+var MessageSubscription = require('./MessageSubscription');
+
+var ChatAppEventProcessor = function (router, modelId) {
+    esp.model.DisposableBase.call(this);
+    this.router = router;
+    this.modelId = modelId;
+    this.messageSubscription = null;
+};
+
+ChatAppEventProcessor.prototype = Object.create(esp.model.DisposableBase.prototype);
+
+ChatAppEventProcessor.prototype.start = function () {
+    this.messageSubscription = new MessageSubscription(this.router, this.modelId);
+    this.addDisposable(this.messageSubscription);
+    this.observeInitEvent();
+    this.observeThreadSelected();
+    this.observeMessagesReceived();
+};
+
+ChatAppEventProcessor.prototype.observeInitEvent = function () {
+    this.addDisposable(this.router
+        .getEventObservable(this.modelId, "initEvent")
+        .observe(function () {
+            this.messageSubscription.start();
+        }.bind(this))
```

Comment:
How does `.bind(this)` work on a prototype definition? Just not sure how it works :